src/lib_json/json_value.cpp

```diff
@@ -1495,6 +1495,14 @@ Value::iterator Value::end() {
   return iterator();
 }
 
+Value const& Value::front() const { return *begin(); }
```

Comment:
Inline these I think.

Reply:
Don't think I can inline these, since they use an incomplete type (the iterators are only forward declared in the header).
I feel like they should be able to be `noexcept` though, but since neither `begin()` nor `end()` are defined `noexcept` I'm not totally sure that's valid, tbh.

Reply:
Ah. I didn't realize that about the iterators. They're defined at the bottom of value.h, so the inlining could still happen, but it have to be a the bottom of the file where `swap` is defined. But it wouldn't do us a whole lot of good unless `begin` and `end` were also inlined, so nevermind I guess.

Reply:
Ah, didn't notice that! I moved them into the header file and underneath swap now, so that, if someone decides to inline `begin` and `end` at some point, there'll be less friction!